src/Microsoft.ML.Vision/DnnRetrainTransform.cs

```diff
@@ -1155,13 +1156,13 @@ private Tensor CastDataAndReturnAsTensor(T[] data)
                     return new Tensor((double[])(object)data, _dims, TF_DataType.TF_DOUBLE);
                 else if (typeof(T) == typeof(ReadOnlyMemory<char>))
                 {
-                    byte[][] bytes = new byte[_vBuffer.Length][];
-                    for (int i = 0; i < bytes.Length; i++)
+                    string[] strings = new string[_vBuffer.Length];
+                    for (int i = 0; i < strings.Length; i++)
                     {
-                        bytes[i] = Encoding.UTF8.GetBytes(((ReadOnlyMemory<char>)(object)data[i]).ToArray());
+                        strings[i] = new string(((ReadOnlyMemory<char>)(object)data[i]).ToArray());
```

Comment:
Maybe it is time to refactor this into a helper method so it isn't duplicated 3 times?

Reply:
Sure